File: IsSubsequence/IsSubsequence.java
package IsSubsequence;
public class IsSubsequence {
    public static void main(String[] args) {
        String s = "abc", t = "ahbgdc";
        System.out.println(solution(s,t));
    }
    public static boolean solution(String s, String t) {
        int i = 0, j = 0;

        while (i < s.length() && j < t.length()) {
            if (s.charAt(i) == t.charAt(j)) {
                i++; 
            }
            j++; 
        }

        return i == s.length();  
    }
}

File: IsSubsequence/IsSubsequence.py
def isSubsequence(s: str, t: str) -> bool:
    i = 0  
    j = 0  

    while i < len(s) and j < len(t):
        if s[i] == t[j]:
            i += 1  
        j += 1  

    return i == len(s)  

s = "abc"
t = "ahbgdc"
print(isSubsequence(s,t))